backend/src/app.module.ts

```diff
@@ -2,10 +2,25 @@ import { Module } from '@nestjs/common';
 import { AppController } from './app.controller';
 import { ShortenerModule } from './shortener/shortener.module';
 import { AppCacheModule } from './cache/cache.module';
+import { ThrottlerModule } from '@nestjs/throttler';
+
 import { AppConfigModule } from './config/config.module';
+import { AppConfigService } from './config/config.service';
 
 @Module({
-  imports: [AppConfigModule, AppCacheModule, ShortenerModule],
+  imports: [
+    AppConfigModule,
+    AppCacheModule,
+    ThrottlerModule.forRootAsync({
+      imports: [AppConfigModule],
```

Comment:
You don't need to import AppConfigModule twice because it's already a global module.
You can remove the imports array:

```js
ThrottlerModule.forRootAsync({
      inject: [AppConfigService],
      useFactory: (config: AppConfigService) => ({
        ttl: config.getConfig().rateLimit.ttl,
        limit: config.getConfig().rateLimit.limit,
      }),
  }),
```

Reply:
Completely right. Thank you for point it out 😄

Reply:
That's why 4 eyes are better than 2 👀